ui/apps/platform/src/Containers/Violations/Details/Deployment/PortConfiguration.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { ReactElement } from 'react';
+import { Card, CardBody, CardTitle, Title } from '@patternfly/react-core';
+
+import { Deployment } from 'types/deployment.proto';
+import PortDescriptionList from './PortDescriptionList';
+
+export type PortConfigurationProps = {
+    deployment: Deployment | null;
+};
+
+function PortConfiguration({ deployment }: PortConfigurationProps): ReactElement {
+    let content: JSX.Element[] | string = 'None';
+
+    if (deployment === null) {
+        content =
+            "Port configurations are unavailable because the alert's deployment no longer exists.";
+    } else {
+        if (deployment.ports.length !== 0) {
```

Comment:
Nit: this could be collapsed into an `else if` to avoid an extra level of indentation.